### DISCUSSION TITLE
Call destructor of vidsrc_st in custom module

### DISCUSSION THREAD

**Discussion by LuThaiHoc** [baresip/baresip#2199] [General]
I'm writing my own video source module with special API of camera. I was successed on send frame buffer from vidsrc to vidisp (test with /vidloop) but the program going crash when I stop video loop (/vidloop_stop) cause by the custom API video thread still send buffer via "frameh(&f, timestamp, arg)"
the function belike:
```
vidsrc_alloc_h{
     startCamera() //realtime push frame buff to camera_frame_handler function
}
```

```
void camera_frame_handler(buffer){
    //process data buffer code
    //
    vidframe_init_buf(&f, VID_FMT_YUV420P, size, buff);
    frameh(&f, timestamp, arg); // send vidframe to viddisp
}
```
When I stop the video loop (/vidloop_stop) the camera still running (I have no idea to call destroy function to stop camera like others vidsrc modules (v4l2, cairo)
I wonder if there is a function to know that video loop has stopped to stop camera and get out of sending video buff to viddisp or how can I call destroy function in vidsrc (whenever call terminated or input /vidloop_stop)
I hope my camera frame handler function belike:
```
void camera_frame_handler(buffer){
    if (vidloop has stopped){
        stopCamera();
        return;
    }
    //process data code
    vidframe_init_buf(&f, VID_FMT_YUV420P, size, buff);
    frameh(&f, timestamp, arg);
}
```
or in alloc function:
```
vidsrc_alloc_h{
     startCamera() //realtime push frame buff to camera_frame_handler function
     // need register **destroy_function** to stop camera when stop vidloop
}
```

**Comment by alfredh** [comment]
can you please share your code and/or logs ?
